lib/src/server/server.dart

```diff
@@ -320,15 +325,24 @@ class Server extends ConnectionServer {
   }
 
   Future<void> shutdown() async {
-    final done = _connections.map((connection) => connection.finish()).toList();
-    if (_insecureServer != null) {
-      done.add(_insecureServer!.close());
-    }
-    if (_secureServer != null) {
-      done.add(_secureServer!.close());
-    }
-    await Future.wait(done);
+    await Future.wait([
+      for (var connection in _connections) connection.finish(),
+      if (_insecureServer != null) _insecureServer!.close(),
+      if (_secureServer != null) _secureServer!.close(),
+    ]);
     _insecureServer = null;
     _secureServer = null;
   }
 }
+
+extension on Socket {
+  InternetAddress? get remoteAddressOrNull {
+    try {
+      // Using a try-catch control flow as dart:io Sockets don't expose their
+      // connectivity state.
+      return remoteAddress;
+    } on Exception catch (_) {
```

Comment:
This is good. My fix had SocketException, which is what remoteAddress is documented to throw, but it actually throws an OSError.  Despite its name, OSError is an Exception, not an Error, so this code is right.